Lib/traceback.py

```diff
@@ -328,13 +332,27 @@ def _original_line(self):
         self.line
         return self._line
 
+    @property
+    def _dedented_lines(self):
+        # Returns _original_line, but dedented
+        self.line
+        if self._line_dedented is None:
+            if self._line is not None:
+                self._line_dedented = textwrap.dedent(self._line).rstrip()
+        return self._line_dedented
+
     @property
     def line(self):
         if self._line is None:
             if self.lineno is None:
                 return None
-            self._line = linecache.getline(self.filename, self.lineno)
-        return self._line.strip()
+            end_lineno = self.lineno if self.end_lineno is None else self.end_lineno
+            self._line = ""
+            for lineno in range(self.lineno, end_lineno + 1):
+                # treat errors and empty lines as the same
+                self._line += linecache.getline(self.filename, lineno).rstrip() + "\n"
```

Comment:
Nit: not that it matters, but this can be cuadratic. Better to place the lines in a list and join at the end. This also helps with debugging.